Week 04/Lecture 07/Assignment 3/Code/lab1/src/main/java/jebi/hendardi/assignment/service/EmployeeService.java

```diff
@@ -0,0 +1,46 @@
+package jebi.hendardi.assignment.service;
+
+import jebi.hendardi.assignment.service.EmailService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+
+@Service
+@Scope("singleton")
+public class EmployeeService {
+
+    private final EmailService emailService;
+
+    // Constructor-based injection
+    @Autowired
```

Comment:
Can we remove @Autowired ? Pls explain

Reply:
Yes, can remove @Autowired from the constructor of EmployeeService where EmailService is injected. 

Spring will still correctly inject EmailService into the constructor because it's the only constructor available for injection and Spring can automatically detect it.

Since this is the sole constructor, Spring will automatically understand that it should inject EmailService here, even without the @Autowired annotation explicitly present.

I already delete @Autowired from this Constructor-based injection.

Reply:
really good